apps/service/src/utils/case.ts

```diff
@@ -0,0 +1,23 @@
+import { SnakeCasedPropertiesDeep, CamelCasedPropertiesDeep } from 'type-fest';
+import { toSnake, toCamel } from 'convert-keys';
+
+export type SnakeCased<T> = SnakeCasedPropertiesDeep<T>;
+export type CamelCased<T> = CamelCasedPropertiesDeep<T>;
+
+export const toSnakeCase = <T>(obj: T): SnakeCased<T> | null => {
+  try {
+    return toSnake(obj);
+  } catch (error) {
+    console.error(error);
+  }
+  return null;
+};
+
+export const toCamelCase = <T>(obj: T): CamelCased<T> | null => {
```

Comment:
The `toCamelcase` of `ckb` lib can convert data to camel case with deep mode and I can rename the function to `toCamelCase`.

Reply:
I think using third-party packages for the feature is acceptable. If you're updating it, the return type of the `toCamelcase` function in the ckb lib could be refactored to be more accurate, using the `CamelCased<T>` type instead of `T` as the return type.